### PR TITLE
Accommodate new OMPL versions which use std smart pointers

### PR DESCRIPTION
**Pull request by cdellin**
The OMPL on my system (ROS kinetic, version 1.2.0) uses std smart pointers, instead of boost smart pointers.  This PR contains a configure-time check to determine which version of OMPL is being built against, and some ifdefs so that `or_ompl` works against either version.  I tested this on my machine, as well as on a machine with the older version of OMPL.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,6 +39,11 @@ catkin_package(
     DEPENDS Boost Eigen OMPL OpenRAVE
 )
 
+if(OR_OMPL_HAS_BOOSTSMARTPTRS)
+else(OR_OMPL_HAS_BOOSTSMARTPTRS)
+add_definitions(-std=c++11)
```
Comment (mkoval) [inline CMakeLists.txt]: You should only use `add_definitions` for `-D` flags. Please add this to `CMAKE_CXX_FLAGS` instead.

File: include/or_ompl/StateSpaces.h [hunk 2]
```diff
@@ -94,7 +99,11 @@ class ContinuousJointsProjectionEvaluator : public ompl::base::ProjectionEvaluat
         ompl::base::ProjectionMatrix _projectionMatrix;
 };
 
+#ifdef OR_OMPL_HAS_BOOSTSMARTPTRS
 typedef boost::shared_ptr<ContinuousJointsProjectionEvaluator> ContinuousJointsProjectionEvaluatorPtr;
+#else
+typedef std::shared_ptr<ContinuousJointsProjectionEvaluator> ContinuousJointsProjectionEvaluatorPtr;
+#endif
```
Comment (mkoval) [inline include/or_ompl/StateSpaces.h]: Please combine these into a single `#ifdef` block, ideally in a dedicated `.h` file so we don't clutter up the code with scattered definitions.

File: CMakeLists.txt (hunk 1 above)
Comment (mkoval) [inline CMakeLists.txt]: Actually - you are using these flags in headers that may be included by other projects. Please use `configure_file` to generate a `.h` file with a `#cmakedefine` in it. That way the variable will be set for other projects that include `or_ompl`'s headers.

Comment (cdellin) [inline CMakeLists.txt]: Fixed! And the `config.h` file was already being generated, so I had just added the new variable to it.

File: include/or_ompl/StateSpaces.h (hunk 2 above)
Comment (cdellin) [inline include/or_ompl/StateSpaces.h]: Yup, I coalesced these at the bottom of this file.

**Comment by cdellin** [comment]
Comments addressed.

File: CMakeLists.txt [hunk 3]
```diff
@@ -30,6 +39,11 @@ catkin_package(
     DEPENDS Boost Eigen OMPL OpenRAVE
 )
 
+if(OR_OMPL_HAS_BOOSTSMARTPTRS)
+else(OR_OMPL_HAS_BOOSTSMARTPTRS)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+endif(OR_OMPL_HAS_BOOSTSMARTPTRS)
```
Comment (mkoval) [inline CMakeLists.txt]: Please replace this with `if (NOT OR_OMPL_HAS_BOOSTSMARTPTRS)`.

**Comment by mkoval** [comment]
Looks good to me besides that one formatting issue. I'll merge once you address that.

File: CMakeLists.txt (hunk 3 above)
Comment (cdellin) [inline CMakeLists.txt]: Done!

**Comment by cdellin** [comment]
CMakeLists formatting addressed.

**Comment by cdellin** [comment]
Thanks for the further formatting fix!

**Pull request merged**